### PR TITLE
mgr/cephadm: introduce flake8

### PR DESCRIPTION
**Pull request by mgfritch**
enable flake8 as a linter for the cephadm module

Signed-off-by: Michael Fritch <mfritch@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/tox.ini [hunk 1]
```diff
@@ -4,18 +4,27 @@ envlist =
...
 
 [flake8]
 max-line-length = 100
+ignore =
+    E501,
+    F241,
+    F401,
+    F811,
+    F841,
+    W503,
```
Comment (mgfritch) [inline src/pybind/mgr/tox.ini:19]: Ignoring these for now since they are more involved to review/fix. Will follow-up with changes in a separate PR.

**Review by jmolmo — APPROVED** [review]

File: src/pybind/mgr/cephadm/module.py [hunk 2]
```diff
@@ -1176,8 +1176,8 @@ def _hosts_with_daemon_inventory(self) -> List[HostSpec]:
         """
         return [
             h for h in self.inventory.all_specs()
-            if self.cache.host_had_daemon_refresh(h.hostname) and
-            h.status.lower() not in ['maintenance', 'offline']
+            if self.cache.host_had_daemon_refresh(h.hostname)
+            and h.status.lower() not in ['maintenance', 'offline']
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/module.py:1180]: the in indentation is wrong. Line 1180 should be indented as we're in the `if` block

**Review by sebastian-philipp — CHANGES_REQUESTED** [review]

**Comment by sebastian-philipp** [comment]
jenkins test make check

File: src/pybind/mgr/cephadm/module.py (hunk 2 above)
Comment (mgfritch) [reply src/pybind/mgr/cephadm/module.py:1180]: hmm it looks wrong to me too, but `autopep8 --select W504` insists this is the correct formatting.

likewise adding the extra indention causes this:
```
cephadm/module.py:1180:17: E131 continuation line unaligned for hanging indent
```

Comment (mgfritch) [reply src/pybind/mgr/cephadm/module.py:1180]: maybe we refactor this block?

```diff
diff --git a/src/pybind/mgr/cephadm/module.py b/src/pybind/mgr/cephadm/module.py
index 7be272daff7..1283a6ae3c9 100644
--- a/src/pybind/mgr/cephadm/module.py
+++ b/src/pybind/mgr/cephadm/module.py
@@ -1174,11 +1174,12 @@ class CephadmOrchestrator(orchestrator.Orchestrator, MgrModule,
         ``_apply_all_specs()`` was called. thus we end up with a hosts
         where daemons might be running, but we have not yet detected them.
         """
-        return [
-            h for h in self.inventory.all_specs()
-            if self.cache.host_had_daemon_refresh(h.hostname) and
-            h.status.lower() not in ['maintenance', 'offline']
-        ]
+        host_specs: List[HostSpec] = []
+        for h in self.inventory.all_specs():
+            if self.cache.host_had_daemon_refresh(h.hostname) \
+                    and h.status.lower() not in ['maintenance', 'offline']:
+                host_specs.append(h)
+        return host_specs
```

... but not sure that's any more readable. wdyt?

**Comment by mgfritch** [comment]
* https://pulpito.ceph.com/mgfritch-2021-02-04_03:37:11-rados:cephadm-wip-mgfritch-testing-2021-02-03-1624-distro-basic-smithi/
* https://pulpito.ceph.com/mgfritch-2021-02-06_22:19:46-rados:cephadm-wip-mgfritch-testing-2021-02-06-1400-distro-basic-smithi/

e2e failures:
https://tracker.ceph.com/issues/49205

**Review by sebastian-philipp — APPROVED** [review]

**Pull request merged**